app/controllers/estimates_controller.rb

```diff
@@ -4,32 +4,34 @@ def new
   end
 
   def create
-    @model = CfeService.call(cfe_session_data(:estimate_id))
-    track_page_view(assessment_id: params[:estimate_id], page: :view_results)
-    render :show
+    redirect_to estimate_path(params[:estimate_id])
+  end
+
+  def show
+    @model = CfeService.call(session_data)
+    track_page_view(page: :view_results)
   end
 
   def print
-    @model = CfeService.call(cfe_session_data(:id))
-    @answers = CheckAnswersPresenter.new cfe_session_data(:id)
-    track_page_view(assessment_id: params[:id], page: :print_results)
+    @model = CfeService.call(session_data)
+    @answers = CheckAnswersPresenter.new session_data
+    track_page_view(page: :print_results)
     render :print, layout: "print_application"
   end
 
   def check_answers
-    @answers = CheckAnswersPresenter.new cfe_session_data(:id)
-    @estimate_id = params.fetch(:id)
-    track_page_view(assessment_id: @estimate_id, page: :check_answers)
+    @answers = CheckAnswersPresenter.new session_data
+    track_page_view(page: :check_answers)
   end
 
   def download
-    track_page_view(assessment_id: params[:id], page: :download_results)
+    track_page_view(page: :download_results)
     html = render_to_string({
       template: "estimates/print",
       layout: "print_application",
       locals: {
-        :@model => CfeService.call(cfe_session_data(:id)),
-        :@answers => CheckAnswersPresenter.new(cfe_session_data(:id)),
+        :@model => CfeService.call(session_data),
+        :@answers => CheckAnswersPresenter.new(session_data),
```

Comment:
This code generated warnings for me about being non rails 7.1 compatible. Not sure how to fix it through? Do we just set local variables?

Reply:
I'll experiment and find out.